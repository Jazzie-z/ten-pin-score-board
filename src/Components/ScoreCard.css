.outer__wrap {
  display: inline-block;
}
.card__wrapper {
  background: #fb4747;
  color: white;
  display: inline-block;
  font-size: 1em;
  font-weight: bold;
  position: relative;
  width: 4.5em;
  white-space: pre;
  box-sizing: border-box;
  border-left: 0.125em solid #a20f0f;
}
.first__roll {
  padding: 0.312em 0 0.312em 0.938em;
  height: 1.313em;
}

.inner__card {
  background: #e1e1e1;
  color: black;
  height: 1.313em;
  width: 1.875em;
  position: absolute;
  text-align: center;
  padding: 0.125em;
  right: 0;
  top: 0;
}

.frame__score {
  text-align: center;
  padding: 0.312em 0;
  height: 1.313em;
}

h2 {
  text-align: center;
}
