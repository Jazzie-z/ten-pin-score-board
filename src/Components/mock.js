export const mockRolls = [
  { primary: "1", secondary: "0" },
  { primary: "5", secondary: "5" },
  { primary: "4", secondary: "0" }
];

export const mockTotal = [1, 15, 19];
