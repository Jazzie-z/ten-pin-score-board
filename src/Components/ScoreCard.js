import React from "react";
import "./ScoreCard.css";

function ScoreCard({ score, total, index }) {
  return (
    <div className={"outer__wrap"}>
      <h2>{index}</h2>
      <div className={"card__wrapper"}>
        <div className={"first__roll"}>{score.primary}</div>
        <div className={"inner__card "}>{score.secondary}</div>
        <div className={"frame__score"}>{total >= 0 ? total : " "}</div>
      </div>
    </div>
  );
}

ScoreCard.defaultProps = {
  score: {}
};
export default ScoreCard;
