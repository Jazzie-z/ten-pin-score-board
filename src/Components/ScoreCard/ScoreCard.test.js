import React from "react";
import { render } from "../../setupTest";
import ScoreCard from "./ScoreCard";
import { mockRolls } from "../mock";

describe("ScoreCard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(<ScoreCard score={mockRolls[0]} />);
  });
  test("snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders with value", () => {
    expect(wrapper.find(".first__roll").text()).toBe(mockRolls[0].primary);
  });

  it("renders without value", () => {
    wrapper = render(<ScoreCard score={{}} total={0} />);
    expect(wrapper.find(".first__roll").text()).toBe("");
  });
});
