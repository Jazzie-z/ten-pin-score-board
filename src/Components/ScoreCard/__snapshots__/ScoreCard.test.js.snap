// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard component snapshot test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "outer__wrap",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "h2",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "card__wrapper",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "first__roll",
              },
              "children": Array [
                Object {
                  "data": "1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "inner__card ",
                },
                "children": Array [
                  Object {
                    "data": "0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "frame__score",
                  },
                  "children": Array [
                    Object {
                      "data": " ",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "inner__card ",
              },
              "children": Array [
                Object {
                  "data": "0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "frame__score",
                },
                "children": Array [
                  Object {
                    "data": " ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "first__roll",
                },
                "children": Array [
                  Object {
                    "data": "1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "frame__score",
              },
              "children": Array [
                Object {
                  "data": " ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "inner__card ",
                },
                "children": Array [
                  Object {
                    "data": "0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "first__roll",
                  },
                  "children": Array [
                    Object {
                      "data": "1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "card__wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "first__roll",
            },
            "children": Array [
              Object {
                "data": "1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "inner__card ",
              },
              "children": Array [
                Object {
                  "data": "0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "frame__score",
                },
                "children": Array [
                  Object {
                    "data": " ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "inner__card ",
            },
            "children": Array [
              Object {
                "data": "0",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "frame__score",
              },
              "children": Array [
                Object {
                  "data": " ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "first__roll",
              },
              "children": Array [
                Object {
                  "data": "1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "frame__score",
            },
            "children": Array [
              Object {
                "data": " ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "inner__card ",
              },
              "children": Array [
                Object {
                  "data": "0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "first__roll",
                },
                "children": Array [
                  Object {
                    "data": "1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "h2",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
